### PR TITLE
Readme update for k8s minor version overlays

### PR DESCRIPTION
**Pull request by saikat-royc**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
/kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Readme update for k8s minor version overlays
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,12 +19,21 @@ Also see [known issues](KNOWN_ISSUES.md) and [CHANGELOG](CHANGELOG.md).
...
 
 | Filestore CSI Driver\Kubernetes Version | 1.14 | 1.15 | 1.16 | 1.17+ |
 | --------------------------------------- | ---- | ---- | ---- | ----- |
 | v0.2.0 (alpha)                          | yes  |  no  |  no  |  no   |
-| v0.3.1 (beta)                           | no   |  no  |  no  |  yes  |
-| master                                  | no   |  no  |  no  |  yes  |
+| v0.3.1 (beta)                           | no   |  yes |  yes |  yes  |
+| master                                  | no   |  yes |  yes |  yes  |
+
+The manifest bundle which captures all the driver components (driver pod which includes the containers csi-external-provisioner, csi-external-resizer, csi-external-snapshotter, gcp-filestore-driver, csi-driver-registrar;
```
Comment (mattcary) [inline README.md:31]: maybe give a bit more detail about what's going on, eg saying that not all driver components can be used with all kubernetes versions (eg resizer, snapshotter), and so different overlays must be used to run v0.3.1 on different k8s versions.

Comment (saikat-royc) [reply README.md:31]: Done. added an example of snapshotter sidecar.

File: test/run-k8s-integration.sh [hunk 2]
```diff
@@ -12,7 +12,7 @@ if [ "$K8S_E2E_SCRIPT_PARENT_DIR" != "$PKGDIR" ]; then
   exit 1
 fi
 
-readonly overlay_name="${GCE_FS_OVERLAY_NAME:-stable}"
+readonly overlay_name="${GCE_FS_OVERLAY_NAME:-stable-master}"
```
Comment (saikat-royc) [inline test/run-k8s-integration.sh:15]: with this change, [pull-gcp-filestore-csi-driver-kubernetes-integration](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/gcp-filestore-csi-driver/gcp-filestore-csi-driver-config.yaml#L71) will start running test against stable-master

**Pull request merged**